### PR TITLE
Add command line flag to enable running examples from file name

### LINKED ISSUE #484: Refactor RunExample to run self-contained examples
Refactor RunExample https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/c1bda42985936a4b49e590e0b3227744dbf48c1f/pkg/client/example-runner.go#L60

to change the signature from what it currently is:

```
func RunExample(client *Client, example broker.ServiceExample, service *broker.ServiceDefinition) error {
```

to instead be:

```
func RunExample(client *Client, example CompleteServiceExample) error {

```

and define a new struct `CompleteServiceExample` that includes enough information from the service to run the example (for instance, the current ServiceExamples don't include the service GUID which is necessary if you wanted to execute the example without knowing which service they belonged to)

This will unblock the ability to create a REST endpoint to fetch examples from and feed back to a service-broker.

### PR DESCRIPTION
**Pull request by sophiawho**
## Description
This commit further resolves #484 by enabling clients to run their own examples against the gcp service broker by providing a `.json` file that contains a list of `CompleteServiceExample` structs. Clients can still specify a specific `serviceName` or `exampleName` along with the file name.

To run the service broker with the file name flag:
```
    ./gcp-service-broker --config config.yml client run-examples --service-examples-file ${FILENAME.json}
```

## Testing
1. An example of how to pretty-print `CompleteServiceExamples` is given in the `ExampleGetAllCompleteServiceExamples2()` func in `example-runner_test.go`.

2. There is a quick sanity check example for running service examples given a file name in the `ExampleRunExamplesFromFile()` func in `example-runner_test.go`. An example file, `complete-service-examples-testfile.json` is given in the same directory.

## Minor Details
1. Moves logic that checks for the error case of a specified `exampleName` and no specified `serviceName` to `client.go` to prevent writing the same logic in two separate functions.
2. Minor indentation nits.

### CODE REVIEW

File: pkg/broker/variables.go [hunk 1]
```diff
@@ -161,7 +165,7 @@ func ValidateVariables(parameters map[string]interface{}, variables []BrokerVari
 	return allErrors
 }
 
-func createJsonSchema(schemaVariables []BrokerVariable) map[string]interface{} {
+func CreateJsonSchema(schemaVariables []BrokerVariable) map[string]interface{} {
```
Comment (josephlewis42) [inline pkg/broker/variables.go]: nit: now that this is public, can we add a docstring to it?

File: pkg/client/complete-service-examples-testfile.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[
```
Comment (josephlewis42) [inline pkg/client/complete-service-examples-testfile.json]: test fixtures should be put under a `testdata/` directory in the same package as the test--this should be `pkg/client/testdata/complete-service-examples-testfile.json`

It's a very undocumented feature of go: https://medium.com/@povilasve/go-advanced-tips-tricks-a872503ac859

File: pkg/client/example-runner_test.go [hunk 3]
```diff
@@ -17,6 +21,79 @@ func ExampleGetAllCompleteServiceExamples() {
 	// <nil>
 }
 
+func ExampleGetAllCompleteServiceExamples2() {
```
Comment (josephlewis42) [inline pkg/client/example-runner_test.go]: If you want more than one example against a given type, use an underscore then a camelCase description heading to make it show up in godoc e.g. `ExampleGetAllCompleteServiceExamples_jsonSpec`

File: pkg/client/example-runner_test.go [hunk 4]
```diff
@@ -17,6 +21,79 @@ func ExampleGetAllCompleteServiceExamples() {
...
+	//			"required": [
+	//				"Email"
+	//			],
+	//			"type": "object"
+	//		}
+	//	}
+	//]
+}
+
+func ExampleRunExamplesFromFile() {
```
Comment (josephlewis42) [inline pkg/client/example-runner_test.go]: Will this work if you aren't running the server?

File: pkg/client/complete-service-examples-testfile.json (hunk 2 above)
Comment (sophiawho) [reply pkg/client/complete-service-examples-testfile.json]: Thanks for letting me know!

Comment (sophiawho) [reply pkg/client/complete-service-examples-testfile.json]: Thanks for pointing that out! File moved.

File: pkg/client/example-runner_test.go (hunk 4 above)
Comment (sophiawho) [reply pkg/client/example-runner_test.go]: Test removed.

**Pull request merged**